Library/Homebrew/extend/os/mac/keg.rb

```diff
@@ -31,36 +31,57 @@ def codesign_patched_binary(file)
     return unless Hardware::CPU.arm?
 
     odebug "Codesigning #{file}"
-    # Use quiet_system to squash notifications about resigning binaries
-    # which already have valid signatures.
-    return if quiet_system("codesign", "--sign", "-", "--force",
-                           "--preserve-metadata=entitlements,requirements,flags,runtime",
-                           file)
+    prepare_codesign_writable_files(file) do
+      # Use quiet_system to squash notifications about resigning binaries
+      # which already have valid signatures.
+      return if quiet_system("codesign", "--sign", "-", "--force",
+                             "--preserve-metadata=entitlements,requirements,flags,runtime",
+                             file)
 
-    # If the codesigning fails, it may be a bug in Apple's codesign utility
-    # A known workaround is to copy the file to another inode, then move it back
-    # erasing the previous file. Then sign again.
-    #
-    # TODO: remove this once the bug in Apple's codesign utility is fixed
-    Dir::Tmpname.create("workaround") do |tmppath|
-      FileUtils.cp file, tmppath
-      FileUtils.mv tmppath, file, force: true
+      # If the codesigning fails, it may be a bug in Apple's codesign utility
+      # A known workaround is to copy the file to another inode, then move it back
+      # erasing the previous file. Then sign again.
+      #
+      # TODO: remove this once the bug in Apple's codesign utility is fixed
+      Dir::Tmpname.create("workaround") do |tmppath|
+        FileUtils.cp file, tmppath
+        FileUtils.mv tmppath, file, force: true
+      end
+
+      # Try signing again
+      odebug "Codesigning (2nd try) #{file}"
+      result = system_command("codesign", args: [
+        "--sign", "-", "--force",
+        "--preserve-metadata=entitlements,requirements,flags,runtime",
+        file
+      ], print_stderr: false)
+      return if result.success?
+
+      # If it fails again, error out
+      onoe <<~EOS
+        Failed applying an ad-hoc signature to #{file}:
+        #{result.stderr}
+      EOS
     end
+  end
 
-    # Try signing again
-    odebug "Codesigning (2nd try) #{file}"
+  def prepare_codesign_writable_files(file)
     result = system_command("codesign", args: [
-      "--sign", "-", "--force",
-      "--preserve-metadata=entitlements,requirements,flags,runtime",
-      file
+      "--display", "--file-list", "-", file
     ], print_stderr: false)
```

Comment:
I think
```ruby
result.success?
```
is `false` here whenever the file list is empty.

If so we can probably return early.

Reply:
Is there ever a case where we would expect the file list to be empty? The path of the dylib or executable itself is still included in the ilst.

For instance, for the `chisel` example, the list is:
```
/usr/local/Cellar/chisel/2.0.1/lib/Chisel.framework/Chisel
/usr/local/Cellar/chisel/2.0.1/lib/Chisel.framework/_CodeSignature/CodeResources
```

Reply:
It will be empty if the binary is not signed.